examples/usematches-loader-data/app/useUser.ts

```diff
@@ -0,0 +1,28 @@
+import useLoaderStore from "./useLoaderStore";
+
+interface User {
+  name: string;
+  email: string;
+}
+
+function isUser(user: unknown): user is User {
+  return (
+    !!user &&
+    typeof (user as User).name === "string" &&
+    typeof (user as User).email === "string"
+  );
+}
+
+export default function useUser(): User | undefined {
```

Comment:
When something like this can return `undefined`, I like giving it clear distinction with `Optional`

```suggestion
export default function useOptionalUser(): User | undefined {
```

And then I have another hook (in this case `useUser`) which throws if the value is undefined. That way I can have a component use this hook and not worry about the `undefined` case. Do you think you could put that together?

Reply:
Great point! Pushed the changes!